### PR TITLE
Add Limit to review points

### PR DESCRIPTION
**Pull request by BlackScorp**
### 1. Why is this change necessary?
If you import reviews from external System, you might have more than 5 star rating. Currently the amount of stars which are displayed on a PDP depends only on the points value from Review. if you have a large number in it. you will see many star icons on a PDP

### 2. What does this change do, exactly?
This PR adds a limit to FloatField and sets a limit of maximum 5 points in ProductReviewEntity

### 3. Describe each step to reproduce the issue or behaviour.
1) Find a product with review
2) Edit the points value in the database for a review to a large number e.g. 1000
3) Refresh PDP and show reviews, you will see the 1000 star icons

### 5. Checklist

- [ ] I have rebased my changes to remove merge conflicts
- [x] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/workflow/2020-08-03-implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: src/Core/Framework/DataAbstractionLayer/FieldSerializer/FloatFieldSerializer.php [hunk 1]
```diff
@@ -41,9 +42,14 @@ public function decode(Field $field, $value): ?float
...
     protected function getConstraints(Field $field): array
     {
-        return [
+        $constraints = [
             new NotBlank(),
             new Type('numeric'),
         ];
+
+        if ($field->getMinValue() !== null || $field->getMaxValue() !== null) {
+            $constraints[] = new Range(['min' => $field->getMinValue(), 'max' => $field->getMaxValue()]);
```
Comment (shyim) [inline src/Core/Framework/DataAbstractionLayer/FieldSerializer/FloatFieldSerializer.php:51]: Unit test would be nice for this change

File: changelog/_unreleased/2022-09-17-add-limit-constraints-to-float-field.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+---
+
+# Core
+
+* Extend FloatField with min and max value
+* Extend FloatFieldSerializer with a range constraint
+* Extend Defaults class with new const MAX_REVIEW_RATING_VALUE
+* Set limit to points in ProductReviewEntity 
+
+___
```
Comment (shyim) [inline changelog/_unreleased/2022-09-17-add-limit-constraints-to-float-field.md]: Can you remove all empty stuff

Comment (BlackScorp) [reply changelog/_unreleased/2022-09-17-add-limit-constraints-to-float-field.md]: done

File: src/Core/Framework/DataAbstractionLayer/FieldSerializer/FloatFieldSerializer.php (hunk 1 above)
Comment (BlackScorp) [reply src/Core/Framework/DataAbstractionLayer/FieldSerializer/FloatFieldSerializer.php:51]: done

File: src/Core/Framework/Test/DataAbstractionLayer/Field/FloatFieldTest.php [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+ * @group FloatFieldTest
+ * @internal
+ */
+class FloatFieldTest extends TestCase
+{
+    use KernelTestBehaviour;
+
+    public function testFloatFieldSerializerNullValue(): void
+    {
+        $serializer = $this->getContainer()->get(FloatFieldSerializer::class);
```
Comment (shyim) [inline src/Core/Framework/Test/DataAbstractionLayer/Field/FloatFieldTest.php:28]: Can you make a unit test? Without container and move to tests/unit/

Comment (BlackScorp) [reply src/Core/Framework/Test/DataAbstractionLayer/Field/FloatFieldTest.php:28]: @shyim but the IntFieldSerializer is test that way, i just reused the IntFieldTest

File: src/Core/Framework/Test/DataAbstractionLayer/Field/FloatFieldTest.php [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+use Shopware\Core\Framework\Test\TestCaseBase\KernelTestBehaviour;
+use Shopware\Core\Framework\Validation\WriteConstraintViolationException;
+
+/**
+ * @group FloatFieldTest
+ * @internal
+ */
+class FloatFieldTest extends TestCase
+{
+    use KernelTestBehaviour;
```
Comment (shyim) [inline src/Core/Framework/Test/DataAbstractionLayer/Field/FloatFieldTest.php:24]: Can you get rid of the integration test and move it to tests/unit/

**Comment by shyim** [comment]
Hey,

I will close this pull request as it is inactive for a while. Feel free to create a new one.

**Pull request closed without merging**